provider/core/src/resource.rs

```diff
@@ -403,16 +403,6 @@ impl Writeable for ResourceOptions {
     }
 }
 
-impl From<LanguageIdentifier> for ResourceOptions {
-    /// Create a ResourceOptions with the given language identifier and an empty variant field.
-    fn from(langid: LanguageIdentifier) -> Self {
-        Self {
-            langid: Some(langid),
-            variant: None,
-        }
-    }
-}
-
```

Comment:
In your PR you're changing `ResourceOptions` to wrap a `Locale`, so `impl From<LanguageIdentifier> for ResourceOptions` will have to construct this `Locale`, even if the client already had it but is only passing the `langid` because that's possible. We already have a `LanguageIdentifier -> Locale` conversion, so a `Locale -> ResourceOptions` conversion should be enough if the `ResourceOptions` actually contains a `Locale`.

It is only somewhat tangential, I think we are using `LanguageIdentifier` a lot because it seemed to be the default language code (as it had a macro), but maybe it should actually be `Locale`. Our components seem to all accept `Into<Locale>`.

Reply:
> In your PR you're changing `ResourceOptions` to wrap a `Locale`

My PR is still in draft, and I intend to make it no longer wrap a `Locale` because it unfortunately increased code size by 10-20% relative to LanguageIdentifier. I'll still be using `Locale`, but just not in the data model of `ResourceOptions`

> It is only somewhat tangential, I think we are using `LanguageIdentifier` a lot because it seemed to be the default language code (as it had a macro), but maybe it should actually be `Locale`. Our components seem to all accept `Into<Locale>`.

We recommend `Locale` as the default thing you should use, so I agree that making `locale!()` is nice. I just don't want to make this change yet to `ResourceOptions`.

Reply:
Thanks for rolling this back. I'll resolve the comment.